index.html

```diff
@@ -204,55 +204,1951 @@ <h3>
+            cannot be accessed, the attribute MUST be 0.
+          </p>
+        </dd>
+        <dt>
+          <dfn>productId</dfn> attribute
+        </dt>
+        <dd>
+          <p>
+            A <dfn>product ID</dfn> is an `unsigned short` value assigned by
+            the manufacturer of the device that identifies the product.
+          </p>
+          <p>
+            The {{HIDDevice/productId}} attribute is the [=product ID=] of the
+            device. If the device has no [=product ID=] or the [=product ID=]
+            cannot be accessed, the attribute MUST be 0.
+          </p>
+        </dd>
+        <dt>
+          <dfn>productName</dfn> attribute
+        </dt>
+        <dd>
+          <p>
+            A string identifying the product.
+          </p>
+          <p>
+            For USB HID devices, the <dfn>product name</dfn> SHOULD contain the
+            value of the string descriptor at the index `iProduct` specified in
+            the Device Descriptor. For Bluetooth HID devices, the [=product
+            name=] SHOULD contain the value of the Device Name characteristic.
+            If the device has no [=product name=] or the [=product name=] is
+            empty or cannot be accessed, the attribute MUST return a non-empty
+            string.
+          </p>
+          <p>
+            The {{HIDDevice/productName}} string SHOULD NOT contain any
+            identifiers like a serial number or Bluetooth device address that
+            could be used to uniquely identify the device.
+          </p>
+        </dd>
+        <dt>
+          <dfn>collections</dfn> attribute
+        </dt>
+        <dd>
+          <p>
+            A sequence of {{HIDCollectionInfo}} representing the [=top-level
+            collections=] in the [=Report descriptor=].
+          </p>
+          <div class="note">
+            <dfn>Top-level collections</dfn> group items that serve similar
+            purposes within the device. Applications look at the [=HID Usage=]
+            applied to [=top-level collections=] to identify devices. An
+            application may not recognize a device if it does not follow the
+            conventions for a particular type of device. Refer to the <a href=
+            "https://usb.org/document-library/hid-usage-tables-122">HID Usage
+            Tables</a> document for more information about the conventional
+            behavior for standard HID devices.
+          </div>
+        </dd>
+      </dl>
+      <p>
+        When an input report is received from a |device:HIDDevice|, run the
+        following steps:
+      </p>
+      <ol>
+        <li>If the input report is a [=blocked report=], abort these steps.
+        </li>
+        <li>Let |reportId:octet| be the report ID for this report, or 0 if
+        report IDs are not used.
+        </li>
+        <li>Let |data:DataView| be the data for this report, excluding the
+        report ID byte if report IDs are used.
```

Comment:
This should create a new DataView over the byte sequence representing the report and, in particular, specify whether the report ID is part of that byte sequence.

Reply:
Done